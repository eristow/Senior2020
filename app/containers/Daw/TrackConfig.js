import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';
import { useInjectReducer } from 'utils/injectReducer';

import Slider from 'components/Slider';
import { makeSelectTrackVol } from './selectors';
import { changeTrackVol } from './actions';
import reducer from './reducer';

const Container = styled.div`
  width: 100px;
  background: darkgray;
  border: 2px black;
  border-radius: 5px;
  margin-bottom: 1px;
`;

const Title = styled.p`
  user-select: none;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  display: inline-block;
  margin: 0.1em 0.5em;
`;

const key = 'daw';

export function TrackConfig({ onChangeVol, name, vol }) {
  useInjectReducer({ key, reducer });

  return (
    <Container>
      <Title>{name}</Title>
      <div>
        <Slider
          onChange={e => onChangeVol(name, e)}
          min={-60}
          max={0}
          value={vol[name]}
          width={50}
        />
      </div>
    </Container>
  );
}

TrackConfig.propTypes = {
  onChangeVol: PropTypes.func,
  name: PropTypes.string,
  vol: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  vol: makeSelectTrackVol(),
});

const mapDispatchToProps = dispatch => ({
  onChangeVol: (name, e) => {
    dispatch(changeTrackVol(name, e));
  },
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(withConnect)(TrackConfig);
