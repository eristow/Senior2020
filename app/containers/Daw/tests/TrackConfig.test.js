import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';

import { TrackConfig } from '../TrackConfig';

describe('<TrackConfig />', () => {
  let store;
  const vol = [0, 0, 0, 0];
  const buffer = { volume: { value: 0 } };
  const trackNames = ['Track 1', 'Track 2', 'Track 3', 'Track 4'];

  beforeAll(() => {
    store = configureStore();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();

    render(
      <Provider store={store}>
        <TrackConfig
          dispatch={dispatch}
          vol={vol}
          buffer={buffer}
          trackNames={trackNames}
        />
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <TrackConfig vol={vol} buffer={buffer} trackNames={trackNames} />
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
