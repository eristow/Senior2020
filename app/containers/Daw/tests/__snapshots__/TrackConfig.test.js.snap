// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackConfig /> Should render and match the snapshot 1`] = `
<div
  class="TrackConfig__Container-ouodsg-0 ddcwJq"
>
  <input
    class="InputText-wtwjzw-0 dmSRq"
    font-size="0.75em"
    transform="uppercase"
    type="text"
    value=""
    width="97%"
  />
  <button
    class="TrackConfig__Select-ouodsg-1 cstDwF"
    type="button"
  >
    Select
  </button>
  <div>
    <div
      style="display: inline-block;"
    >
      <div
        class="rc-slider"
        style="width: 75px; height: 5px; margin: 0.5em 0.5em; z-index: 0;"
      >
        <div
          class="rc-slider-rail"
        />
        <div
          class="rc-slider-track"
          style="left: 0%; width: 100%;"
        />
        <div
          class="rc-slider-step"
        />
        <div
          aria-disabled="false"
          aria-valuemax="0"
          aria-valuemin="-60"
          aria-valuenow="0"
          class="rc-slider-handle"
          role="slider"
          style="left: 100%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="rc-slider-mark"
        />
      </div>
    </div>
  </div>
</div>
`;
